backend/core/src/auth/services/user.service.ts

```diff
@@ -621,7 +621,16 @@ export class UserService {
   }
 
   private static getPublicConfirmationUrl(appUrl: string, user: User) {
-    return `${appUrl}?token=${user.confirmationToken}`
+    const urlObj = new URL(appUrl)
+
+    const redirectUrl = urlObj.searchParams.get("redirectUrl")
+    const listingId = urlObj.searchParams.get("listingId")
+
+    const confirmationUrl =
+      redirectUrl && listingId && process.env.SHOW_MANDATED_ACCOUNTS
```

Comment:
We don't have access to this in the backend, I'm curious how this was working for you? Did you maybe add it locally to the private env file?

Reply:
Yes, sorry forgot to mention that. I added the variable in the core env file

Reply:
I can add it to the template